### ISSUE TITLE
[🐞] Build of supabase-auth-helpers-qwik fails because of TS type error

### ISSUE THREAD

**Issue by byte-barista** [QwikDev/qwik#7146]
### Which component is affected?

Qwik Runtime

### Describe the bug

When I run pnpm build.local during build this error appears:

`> supabase-auth-helpers-qwik@0.0.3 build /Users/nemanjamiljkovic/Desktop/qwik/packages/supabase-auth-helpers-qwik
> vite build --mode lib

vite v5.3.5 building for lib...
✓ 3 modules transformed.
lib/index.qwik.mjs  2.80 kB │ gzip: 0.85 kB
lib/index.qwik.cjs  2.96 kB │ gzip: 0.89 kB
✓ built in 42ms
⚛️  supabase-auth-helpers-qwik

packages/docs/src/repl/worker/repl-dependencies.ts:51:3 - error TS2740: Type '{ isServer: true; isBrowser: false; isDev: false; }' is missing the following properties from type 'typeof import("/Users/nemanjamiljkovic/Desktop/qwik/packages/qwik/dist/core")': getLocale, useServerData, withLocale, $, and 86 more.

51   self.qwikBuild = {
     ~~~~~~~~~~~~~~

Found 2 errors in the same file, starting at: packages/docs/src/repl/worker/repl-dependencies.ts:2

❌ Error: Command failed with exit code 2: tsc -p packages/docs/tsconfig.json`

**I was able to resolve it with:**

```
  self.qwikBuild = {
    isServer: true,
    isBrowser: false,
    isDev: false,
  } as typeof import('@builder.io/qwik');
```


### Reproduction

https://github.com/byte-barista/qwik

### Steps to reproduce

pnpm install && pnpm build.local

### System Info

```shell
System:
    OS: macOS 15.1.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 1.50 GB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 22.11.0 - ~/.nvm/versions/node/v22.11.0/bin/node
    npm: 10.9.0 - ~/.nvm/versions/node/v22.11.0/bin/npm
    pnpm: 9.14.2 - ~/.nvm/versions/node/v22.11.0/bin/pnpm
    Watchman: 2024.12.02.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 131.0.6778.109
    Safari: 18.1.1
  npmPackages:
    @builder.io/partytown: 0.10.2 => 0.10.2 
    @builder.io/qwik: workspace:^ => 1.11.0 
    @builder.io/qwik-city: workspace:^ => 1.11.0 
    typescript: 5.4.5 => 5.4.5 
    undici: * => 6.6.2 
    vite: 5.3.5 => 5.3.5
```


### Additional Information

_No response_

**Comment by gioboa** [comment]
Hi @byte-barista I can't access to your repo, is it private?

**Comment by byte-barista** [comment]
Hi @gioboa, I have tried to reproduce the error on Stackblitz and CodeSandbox, but without success. I can't set up the same environment. I've updated the link to the forked repo of Qwik. Essentially, I cloned the project locally and followed the setup instructions from [Contributing.md](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md), but the ```pnpm build.local``` command fails with a TS error as shown above. I was trying to run ```pnpm docs.dev``` because I'm working on drag-and-drop examples for the Cookbook.
